src/mbgl/style/style_impl.cpp

```diff
@@ -203,7 +204,9 @@ Layer* Style::Impl::addLayer(std::unique_ptr<Layer> layer, optional<std::string>
     }
 
     layer->setObserver(this);
-    observer->onUpdate(Update::Repaint);
+    if (loaded) {
+        observer->onUpdate(Update::Repaint);
```

Comment:
Since the observer is `Map::Impl` here and in `Map::Impl::onUpdate` we check the loaded state again for still requests, this is not needed right?

Same for the other checks.

Reply:
Agreed - we're basically exchanging a single bool check with a vtable lookup + two checks in this particular case, by the cost of maintainability. I'm 👍  in removing the `if (loaded)` checks.